static/js/muting_ui.js

```diff
@@ -158,6 +163,59 @@ export function toggle_topic_mute(message) {
     }
 }
 
+export function persist_user_mute(user_id) {
+    last_user_update = timestamp_ms();
+    channel.post({
+        url: "/json/users/me/muted_users/" + user_id,
+        idempotent: true,
+    });
+}
+
+export function persist_user_unmute(user_id) {
+    last_user_update = timestamp_ms();
+    channel.del({
+        url: "/json/users/me/muted_users/" + user_id,
+        idempotent: true,
+    });
+}
+
 export function handle_user_updates(muted_user_ids) {
+    popovers.hide_user_info_popover();
+    popovers.hide_message_info_popover();
+    popovers.hide_user_sidebar_popover();
+    if (timestamp_ms() < last_user_update + 1000) {
+        return;
+    }
     muting.set_muted_users(muted_user_ids);
 }
+
+export function mute_user(user_id) {
+    popovers.hide_user_info_popover();
+    muting.add_muted_user(user_id);
```

Comment:
It looks like this line is effectively local-echoing the user mute; I think they're be rare enough that it'd probably be better to just not do that sort of local echo (i.e. delete this line and let us get the update via `handle_user_updates` -- I think that'd let us get rid of the 1000 hack at least for this new code).

(Topic edits are really common and have a reasonable case for doing so).